File: C++/Easy/Accepted/unique-number-of-occurrences_7.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        
        map<int, int> count;
        map<int, int> freq;
        for (int i = 0; i < arr.size(); i++) {
            count[arr[i]]++;
        }
        for (auto it = count.begin(); it != count.end(); it++) {
            freq[it->second]++;
        }
        for (auto it = freq.begin(); it != freq.end(); it++) {
            if (it->second > 1) {
                return false;
            }
        }
        return true;
    }
};

File: Rust/Easy/Accepted/unique-number-of-occurrences_7.rs
impl Solution {
    pub fn unique_occurrences(arr: Vec<i32>) -> bool {
        
        use std::collections::HashMap;
        let mut map = HashMap::new();
        for i in &arr {
            *map.entry(*i).or_insert(0) += 1;
        }
        let mut set = std::collections::HashSet::new();
        for i in map.values() {
            if !set.insert(i) {
                return false;
            }
        }
        true
    }
}